pkg/channel/message_dispatcher.go

```diff
@@ -233,9 +234,10 @@ func (d *MessageDispatcherImpl) executeRequest(ctx context.Context,
 	}
 	execInfo.Time = dispatchTime
 
+	body := make([]byte, attributes.KnativeErrorDataExtensionMaxLength)
```

Comment:
Since you touch this area, we noticed that it is a good idea to do base64 encoding of the actual body.

See https://github.com/knative-sandbox/eventing-kafka-broker/pull/2604  and the issue it is fixing

Reply:
Sure, I've added it in the Commit f8a2fa97055ca34a63f4424559650250432a60ef

Reply:
It seems to involve multiple unit tests and e2e tests, is it ok for me to add it to another PR?

Reply:
yes, it is for sure OK to be done in a different PR.

It was more a heads-up!